src/components/NavBar.jsx

```diff
@@ -1,35 +1,29 @@
+import { useContext } from 'react';
+import { GlobalContext } from '../context/GlobalContext';
 import Navbar from 'react-bootstrap/Navbar';
 import Nav from 'react-bootstrap/Nav';
-import Form from 'react-bootstrap/Form';
 import SearchForm from './SearchForm';
-import PropTypes from 'prop-types';
 
-const NavBar = ({ query, updateQuery, getSearch }) => {
+const NavBar = () => {
+  const { darkMode, dispatch } = useContext(GlobalContext);
+  const handleToggle = () => {
+    dispatch({ type: 'TOGGLE_THEME' });
+  };
+
   return (
     <Navbar collapseOnSelect expand="false" bg="success" variant="dark">
-      <SearchForm
-        query={query}
-        updateQuery={updateQuery}
-        getSearch={getSearch}
-      />
+      <SearchForm />
       <Navbar.Toggle aria-controls="responsive-navbar-nav" className="m-2" />
       <Navbar.Collapse id="responsive-navbar-nav">
         <Nav className="d-flex m-2">
           <Nav.Link>Log In</Nav.Link>
-          <Nav.Link className="d-flex">
-            <span className="me-2">Dark Mode</span>
-            <Form.Switch />
+          <Nav.Link onClick={handleToggle}>
+            Activate {darkMode ? 'Light' : 'Dark'} Mode
```

Comment:
I think if you create a component just for the button that switches the theme would be nice. Also a switch component may be better, only do it if there is enough time